### PR TITLE
ledger refactoring: update evaluator benchmarks

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Update the evaluator benchmark to cover application calls.

Note: the prefetcher changes would follow this one.

## Test Plan

These are tests.

### CODE REVIEW

File: ledger/evalbench_test.go [hunk 1]
```diff
@@ -239,10 +287,58 @@ func BenchmarkBlockEvaluatorDiskNoCrypto(b *testing.B) {
...
+		MaxAppsOptedIn:   config.Consensus[protocol.ConsensusV30].MaxAppsOptedIn,
+		TransactionsType: protocol.PaymentTx,
+	}
+	benchmarkBlockEvaluator(b, false, false, protocol.ConsensusFuture, &g)
+}
+
+func BenchmarkBlockEvaluatorDiskAppCalls(b *testing.B) {
+	if b.N < 2 {
+		b.N = 2
+	}
```
Comment (jannotti) [inline ledger/evalbench_test.go:303]: Why is this needed?

File: ledger/evalbench_test.go [hunk 2]
```diff
@@ -97,27 +97,28 @@ func (g *BenchPaymentTxnGenerator) Txn(tb testing.TB, addrs []basics.Address, ke
...
 	maxLocalSchemaEntries := config.Consensus[g.Proto].MaxLocalSchemaEntries
-	maxAppsOptedIn := config.Consensus[g.Proto].MaxAppsOptedIn
-	if maxAppsOptedIn == 0 {
-		maxAppsOptedIn = config.Consensus[protocol.ConsensusV30].MaxAppsOptedIn
-	}
+	maxAppsOptedIn := g.MaxAppsOptedIn
 
-	// this function might create too much transaction even to fit into a single block
+	// this function might create more transactions than a single block could contain.
 	// estimate number of smaller blocks needed in order to set LastValid properly
```
Comment (jannotti) [inline ledger/evalbench_test.go:118]: If that's the only reason for the extra work, why not just set LastValid to FirstValid + MaxTxLifetime?

File: ledger/evalbench_test.go [hunk 3]
```diff
@@ -239,10 +287,58 @@ func BenchmarkBlockEvaluatorDiskNoCrypto(b *testing.B) {
...
 
 func BenchmarkBlockEvaluatorDiskAppOptIns(b *testing.B) {
-	g := BenchAppOptInsTxnGenerator{
-		NumApps: 500,
-		Proto:   protocol.ConsensusFuture,
-		Program: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+	g := benchAppOptInsTxnGenerator{
+		NumApps:          500,
+		Proto:            protocol.ConsensusFuture,
+		Program:          []byte{0x02, 0x20, 0x01, 0x01, 0x22},
```
Comment (jannotti) [inline ledger/evalbench_test.go:293]: It would be nice to assemble this, so the reader knows what it does.

**Review by jannotti — APPROVED** [review]
I don't see any problems.  Just small comments/questions.

File: ledger/evalbench_test.go (hunk 1 above)
Comment (algorandskiy) [reply ledger/evalbench_test.go:308]: Please add a comment about this.
Moreover, `benchmarkBlockEvaluator` uses b.N only in
```
	for i := 0; i < b.N; i++ {
		if withCrypto {
			_, err = l2.Validate(context.Background(), validatedBlock.Block(), backlogPool)
		} else {
			_, err = internal.Eval(context.Background(), l2, validatedBlock.Block(), false, nil, nil)
		}
		require.NoError(b, err)
	}
```
with b.N in range [1,2] does not make much sense.
